### DISCUSSION TITLE
Running a crates.io proxy cache / mirror

### DISCUSSION THREAD

**Discussion by emostov** [rust-lang/crates.io#9230] [Q&A]
Hey! 

My project is getting rate limited from crates.io and I wanted to setup a pull through cache for my team. 

Is there anything off the shelf I can use to setup a self hosted pull through cache?

Appreciate any advice and links to existing ecosystem projects that may serve as an example!

**Comment by Turbo87** [comment]
one option I'm aware of is https://kellnr.io/

for regular downloads from crates.io you shouldn't get rate limited though 🤔
